src/Microsoft.Identity.Web/Resource/RegisterValidAudience.cs

```diff
@@ -61,7 +61,12 @@ internal class RegisterValidAudience
             SecurityToken securityToken,
             TokenValidationParameters validationParameters)
         {
-            JwtSecurityToken token = securityToken as JwtSecurityToken;
+            JwtSecurityToken? token = securityToken as JwtSecurityToken;
+            if (token == null)
+            {
+                throw new SecurityTokenValidationException("Token is not JWT token.");
```

Comment:
Hope this a good exception here.

Reply:
LOL.

---
In reply to: [433752648](https://github.com/AzureAD/microsoft-identity-web/pull/183#discussion_r433752648) [](ancestors = 433752648)